pkg/reconciler/subscription/subscription.go

```diff
@@ -283,7 +315,7 @@ func (r *Reconciler) resolveResult(ctx context.Context, namespace string, replyS
 		logging.FromContext(ctx).Warn("Failed to deserialize Addressable target", zap.Error(err))
 		return "", err
 	}
-	if s.Status.Address != nil {
+	if s.Status.Address != nil && s.Status.Address.Hostname != "" {
```

Comment:
I can't comment on line 292, but:

we should mark the channel not ready when we call `syncPhysicalChannel`. The channel controller needs the chance to say it has read the updated subscribers.

Reply:
@mattmoor

Reply:
This came up as a more general issue, when do things become ready? https://github.com/knative/eventing/issues/1076#issuecomment-488145482

Reply:
When the channel controller reconciles again and gets to decide. Lets do it as a followup.